interpreter/cling/lib/Interpreter/CIFactory.cpp

```diff
@@ -1043,6 +1043,9 @@ static void stringifyPreprocSetting(PreprocessorOptions& PPOpts,
                                           /*UserFilesAreVolatile*/ true);
     CI->setSourceManager(SM); // CI now owns SM
 
+    if (Invocation.getFrontendOpts().ModulesEmbedAllFiles)
+       CI->getSourceManager().setAllFilesAreTransient(true);
```

Comment:
Does this have a perf impact? I remember a logical connection in clang's code between "is the file transient" and "can we cache the file content"?

Reply:
I do not expect such. Esp that we are using it for in rootcling only. That just flips a bit on the FileEntry and the ContentCache to tell the AST reader not to expect on disk files.